pkg/controllers/cluster/taint_manager.go

```diff
@@ -170,6 +171,10 @@ func (tc *NoExecuteTaintManager) syncBindingEviction(key util.QueueKey) error {
 	// Case 2: Need eviction after toleration time. If time is up, do eviction right now.
 	// Case 3: Tolerate forever, we do nothing.
 	if needEviction || tolerationTime == 0 {
+		err := controllerUtils.UpdateFailoverStatus(tc.Client, binding, workv1alpha2.EvictionReasonTaintUntolerated)
```

Comment:
The type of the third input parameter is incorrect. The correct type is workv1alpha2.FailoverReason.